### ISSUE TITLE
Cannot install

### ISSUE THREAD

**Issue by Dundee** [tldr-pages/tldr-node-client#91]
I am not able to install this package. It tells 

    permission denied, mkdir '/root/.tldr/cache'

although ran as root user.

## Output

[root@dundee-dell ~]# npm install -g tldr
npm WARN deprecated graceful-fs@3.0.8: graceful-fs version 3 and before will fail on newer node releases. Please update to graceful-fs@^4.0.0 as soon as possible.
/usr/bin/tldr -> /usr/lib/node_modules/tldr/bin/tldr

> tldr@1.6.0 postinstall /usr/lib/node_modules/tldr
> node ./bin/tldr --update

Updating...
/usr/lib/node_modules/tldr/node_modules/wrench/lib/wrench.js:481
            throw err;
            ^

Error: EACCES: permission denied, mkdir '/root/.tldr/cache'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:842:18)
    at Object.mkdirSyncRecursive (/usr/lib/node_modules/tldr/node_modules/wrench/lib/wrench.js:466:12)
    at /usr/lib/node_modules/tldr/lib/cache.js:55:12
    at Extract.<anonymous> (/usr/lib/node_modules/tldr/lib/remote.js:28:5)
    at emitNone (events.js:85:20)
    at Extract.emit (events.js:179:7)
    at DirWriter.<anonymous> (/usr/lib/node_modules/tldr/node_modules/unzip2/lib/extract.js:42:10)
    at emitNone (events.js:85:20)
    at DirWriter.emit (events.js:179:7)
npm ERR! Linux 4.4.1-2-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install" "-g" "tldr"
npm ERR! node v5.7.0
npm ERR! npm  v3.7.5
npm ERR! code ELIFECYCLE

npm ERR! tldr@1.6.0 postinstall: `node ./bin/tldr --update`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tldr@1.6.0 postinstall script 'node ./bin/tldr --update'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the tldr package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/tldr --update
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs tldr
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls tldr
npm ERR! There is likely additional logging output above.

## npm-debug.log

9935 info lifecycle tldr@1.6.0~postinstall: tldr@1.6.0
9936 verbose lifecycle tldr@1.6.0~postinstall: unsafe-perm in lifecycle false
9937 verbose lifecycle tldr@1.6.0~postinstall: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/usr/lib/node_modules/tldr/node_modules/.bin:/usr/lib/node_modules/.bin:/usr/bin:/usr/lib/hardening-wrapper/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
9938 verbose lifecycle tldr@1.6.0~postinstall: CWD: /usr/lib/node_modules/tldr
9939 silly lifecycle tldr@1.6.0~postinstall: Args: [ '-c', 'node ./bin/tldr --update' ]
9940 silly lifecycle tldr@1.6.0~postinstall: Returned: code: 1  signal: null
9941 info lifecycle tldr@1.6.0~postinstall: Failed to exec postinstall script
9942 verbose unlock done using /root/.npm/_locks/staging-a072192f34a17023.lock for /usr/lib/node_modules/.staging
9943 silly rollbackFailedOptional Starting
9944 silly rollbackFailedOptional Finishing
9945 silly runTopLevelLifecycles Starting
9946 silly runTopLevelLifecycles Finishing
9947 silly install printInstalled
9948 verbose stack Error: tldr@1.6.0 postinstall: `node ./bin/tldr --update`
9948 verbose stack Exit status 1
9948 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:239:16)
9948 verbose stack     at emitTwo (events.js:100:13)
9948 verbose stack     at EventEmitter.emit (events.js:185:7)
9948 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:24:14)
9948 verbose stack     at emitTwo (events.js:100:13)
9948 verbose stack     at ChildProcess.emit (events.js:185:7)
9948 verbose stack     at maybeClose (internal/child_process.js:827:16)
9948 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
9949 verbose pkgid tldr@1.6.0
9950 verbose cwd /root
9951 error Linux 4.4.1-2-ARCH
9952 error argv "/usr/bin/node" "/usr/bin/npm" "install" "-g" "tldr"
9953 error node v5.7.0
9954 error npm  v3.7.5
9955 error code ELIFECYCLE
9956 error tldr@1.6.0 postinstall: `node ./bin/tldr --update`
9956 error Exit status 1

## Environment
- Archlinux (updated), kernel 4.4.1-2-ARCH
- NodeJS 5.7.0, npm 3.7.5

**Comment by rubenvereecken** [comment]
Thanks for the extensive logs! Good news, I managed to reproduce. Now gimme just a second.

**Comment by rubenvereecken** [comment]
Alright so here's the problem. Once tldr is installed, it'll try to fetch the pages and cache them. This post-install script doesn't seem to be executed as root. I'm not entirely sure why it isn't, but you do install in `/usr/lib` which is probably not owned by root, so maybe something happens there.

I tried changing the install path to something owned by root but that didn't fix it either.

Anyone else?

**Comment by igorshubovych** [comment]
You installed the package, but did not update the pages archive.

In your case you can just run `tldr --update` to finish the installation.

We will investigate how to avoid this issue in future.

**Comment by Dundee** [comment]
My ``/usr/lib`` is owned by root. Ownership of files doesn't affect effective user ID of process unless setuid bit is set for the executed file. Maybe this is the case?

**Comment by Dundee** [comment]
It's not installed. The installation was rollbacked because of the failure in postinstall.

**Comment by igorshubovych** [comment]
```bash
npm install -g --ignore-scripts tldr
tldr --update
```

**Comment by Dundee** [comment]
Thanks, that worked.

**Comment by igorshubovych** [comment]
Why you installed it under root privileges?

**Comment by Dundee** [comment]
I suppose it's needed when you install it globally (into /usr/lib).

**Comment by rubenvereecken** [comment]
Ah I was looking for that option, nice one @igorshubovych. Btw, I kind of figured root was your only user. If it isn't and you want a different install path, check out [this](https://docs.npmjs.com/getting-started/fixing-npm-permissions) page.

**Comment by MrPluto** [comment]
@igorshubovych  :+1:  nice

**Comment by igorshubovych** [comment]
I am closing this one. Feel free to reopen.
